vendor/github.com/containers/image/docker/docker_client.go

```diff
@@ -119,6 +120,15 @@ func (c *dockerClient) makeRequestToResolvedURL(method, url string, headers map[
 			return nil, err
 		}
 	}
+	// if the request needs a body from "stream" we set this after we setup authentication
+	// because setupRequestAuth tries the request looking for a Bearer token on Docker Hub
+	// and setting this early will result in the body  being read by client.Do(req) in setupRequestAuth
+	// trying the request in setupRequestAuth doesn't need a body eventually
+	// so we set this here.
+	if stream != nil {
+		req.Body = ioutil.NopCloser(stream)
```

Comment:
The truly clean way to solve this, I suppose, would be to create a new `Request` for the header check in `setupRequestAuth`. Though that might be much more work.